src/camelot/fetchNitroPool.ts

```diff
@@ -0,0 +1,198 @@
+import { BigNumber, ethers } from 'ethers'
+
+import { CamelotNitroPool, ERC20 } from '../typechained'
+import { Geb } from '../geb'
+import { fromBigNumber, multicall, CamelotMulticallRequest, SECONDS_IN_YEAR } from '../utils'
+
+export type NitroPoolDetails = {
+    tvl: number
+    pendingRewards: {
+        pending1: number
+        pending2: number
+    }
+    settings: {
+        startTime: BigNumber
+        endTime: BigNumber
+        harvestStartTime: BigNumber
+        depositEndTime: BigNumber
+        lockDurationReq: BigNumber
+        lockEndReq: BigNumber
+        depositAmountReq: BigNumber
+        whitelist: boolean
+        description: string
+    }
+    rewardsPerSecond: number
+    lpTokenBalance: number
+    userInfo: {
+        totalDepositAmount: BigNumber
+        rewardDebtToken1: BigNumber
+        rewardDebtToken2: BigNumber
+        pendingRewardsToken1: BigNumber
+        pendingRewardsToken2: BigNumber
+    } | null
+    apy: number
+}
+
+type AvailableDepositTypes = 'WSTETH' | 'RETH'
+
+const fetchNitroPool = async (
+    geb: Geb,
+    collateralType: AvailableDepositTypes,
+    address: string | null
+): Promise<NitroPoolDetails> => {
+    const ODGAddress = geb.tokenList['ODG'].address
+    const collateralAddress = geb.tokenList[collateralType].address
+    const collateralChainlinkRelayer = geb.tokenList[collateralType].chainlinkRelayer
+    const odgChainlinkRelayerAddress = geb.tokenList['ODG'].chainlinkRelayer
+
+    if (!ODGAddress || !collateralAddress || !collateralChainlinkRelayer) {
+        console.warn('Missing token info in tokenlist')
+        return {
+            tvl: 0,
+            pendingRewards: {
+                pending1: 0,
+                pending2: 0,
+            },
+            settings: {
+                startTime: BigNumber.from(0),
+                endTime: BigNumber.from(0),
+                harvestStartTime: BigNumber.from(0),
+                depositEndTime: BigNumber.from(0),
+                lockDurationReq: BigNumber.from(0),
+                lockEndReq: BigNumber.from(0),
+                depositAmountReq: BigNumber.from(0),
+                whitelist: false,
+                description: '',
+            },
+            rewardsPerSecond: 0,
+            lpTokenBalance: 0,
+            userInfo: null,
+            apy: 0,
+        }
+    }
+
+    const odg = geb.getErc20Contract(ODGAddress)
+    const collateral = geb.getErc20Contract(collateralAddress)
+
+    const camelotNitroPool = await geb.contracts[`camelot${collateralType}NitroPool`]
+
+    const chainlinkRelayerContract = new ethers.Contract(
```

Comment:
The code to fetch prices should be moved out, since it is also shared with `fetchPoolData` in `tokenData.ts`